File: app/src/main/java/info/maaskant/wouttest2/settings/BindPreferenceSummaryToValueListener.java
package info.maaskant.wouttest2.settings;

import android.preference.ListPreference;
import android.preference.Preference;

/**
 * A preference value change listener that updates the preference's summary to reflect its new
 * value. The exact display format is dependent on the type of preference.
 */
class BindPreferenceSummaryToValueListener implements Preference.OnPreferenceChangeListener {

    @Override
    public boolean onPreferenceChange(Preference preference, Object value) {
        String stringValue = value.toString();

        if (preference instanceof ListPreference) {
            // For list preferences, look up the correct display value in
            // the preference's 'entries' list.
            ListPreference listPreference = (ListPreference) preference;
            int index = listPreference.findIndexOfValue(stringValue);

            // Set the summary to reflect the new value.
            preference.setSummary(index >= 0 ? listPreference.getEntries()[index] : null);

        } else {
            // For all other preferences, set the summary to the value's
            // simple string representation.
            preference.setSummary(stringValue);
        }
        return true;
    }

}

File: app/src/main/java/info/maaskant/wmsnotes/android/ui/settings/BindPreferenceSummaryToValueListener.kt
package info.maaskant.wmsnotes.android.ui.settings

import android.preference.ListPreference
import android.preference.Preference

/**
 * A preference value change listener that updates the preference's summary to reflect its new
 * value. The exact display format is dependent on the type of preference.
 */
internal class BindPreferenceSummaryToValueListener : Preference.OnPreferenceChangeListener {

    override fun onPreferenceChange(preference: Preference, value: Any): Boolean {
        val stringValue = value.toString()

        if (preference is ListPreference) {
            // For list preferences, look up the correct display value in
            // the preference's 'entries' list.
            val index = preference.findIndexOfValue(stringValue)

            // Set the summary to reflect the new value.
            preference.setSummary(if (index >= 0) preference.entries[index] else null)

        } else {
            // For all other preferences, set the summary to the value's
            // simple string representation.
            preference.summary = stringValue
        }
        return true
    }

}
